lib/buildTrunkPackage.nix

```
@@ -0,0 +1,44 @@
+{ mkCargoDerivation
+, trunk
+, wasm-bindgen-cli
+, binaryen
+, nodePackages
+}:
+
+{
+  # this is not called cargoArtifacts to push users
+  # away from giving us useless artifacts built
+  # for native target instead of wasm32-unknown-unknown
+  cargoArtifactsWasm
```

```suggestion
  cargoArtifacts
```

I sympathize with the intent but I'd prefer we call this `cargoArtifacts` since that is the pattern we already use everywhere else (plus various low level helpers know and look for that name).

Yes it is a slight foot gun if someone passes in `cargoArtifacts` related to another target, though it's totally possible to have `cargoArtifacts` consisting of multiple targets too. I'd rather be consistent here and revisit if problems arise

---

Having a second pass on the PR now, how about having an assert in `buildTrunkPackage` checking if `${cargoArtifacts}/target/wasm32-unknown-unknown` is a path?